src/components/Identity/getIdentity/createGetIdentity.js

```diff
@@ -17,11 +17,17 @@ export default ({
   globalConfigOverrides
 }) => {
   return ({ namespaces, edgeConfigOverrides: localConfigOverrides } = {}) => {
-    const payload = createIdentityRequestPayload(namespaces);
-    const request = createIdentityRequest(payload);
+    const requestParams = {};
+    const { edgeConfigId } = localConfigOverrides || {};
+    if (edgeConfigId) {
+      delete localConfigOverrides.edgeConfigId;
+      requestParams.edgeConfigIdOverride = edgeConfigId;
+    }
+    requestParams.payload = createIdentityRequestPayload(namespaces);
+    const request = createIdentityRequest(requestParams);
     // merge the configurations, but give preference to the command-local configs
-    payload.mergeConfigOverride(globalConfigOverrides);
-    payload.mergeConfigOverride(localConfigOverrides);
+    requestParams.payload.mergeConfigOverride(globalConfigOverrides);
+    requestParams.payload.mergeConfigOverride(localConfigOverrides);
     return sendEdgeNetworkRequest({
       request
     });
```

Comment:
I'm not excited about all this duplication. What about a function that does this for you:
```
const buildRequestParams = ({
  localConfigOverrides,
  globalConfigOverrides,
  payload
}) => {
  const requestParams = { payload };
  const { edgeConfigId, ...localConfigOverridesWithoutEdgeConfigId } = localConfigOverrides || {};
  if (edgeConfigId) {
    requestParams.edgeConfigIdOverride = edgeConfigId;
  }
  payload.mergeConfigOverride(globalConfigOverrides);
  payload.mergeConfigOverride(localConfigOverridesWithoutEdgeConfigId);
  return requestParams;
};

export default ({
  sendEdgeNetworkRequest,
  createIdentityRequestPayload,
  createIdentityRequest,
  globalConfigOverrides
 }) => {
  return ({ namespaces, edgeConfigOverrides: localConfigOverrides } = {}) => {
    const request = createIdentityRequest(buildRequestParams({
      localConfigOverrides,
      globalConfigOverrides,
      payload: createIdentityRequestPayload(namespaces)
    }));
    return sendEdgeNetworkRequest({ request });
  };
};
```

Reply:
That's a good idea.